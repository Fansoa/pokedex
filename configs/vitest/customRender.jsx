import { render } from "@testing-library/react";
import IntegrationProvider from "@/configs/vitest/testUtils";

const renderWithIntegrationProvider = (children) =>
  render(<IntegrationProvider>{children}</IntegrationProvider>);

export default renderWithIntegrationProvider;
