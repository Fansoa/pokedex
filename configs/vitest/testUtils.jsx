import { ThemeProvider } from "@emotion/react";
import PropTypes from "prop-types";
import theme from "@/src/theme/theme";

const IntegrationProvider = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

IntegrationProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default IntegrationProvider;
