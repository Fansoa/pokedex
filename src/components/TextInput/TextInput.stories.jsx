import TextInput from "@/src/components/TextInput";

export default {
  title: "Components/TextInput",
  component: TextInput,
};

export const TextInputStory = {
  name: "Default",
};
