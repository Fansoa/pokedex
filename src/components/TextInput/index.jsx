import {
  Container,
  ErrorMessage,
  Input,
  Label,
} from "@/src/components/TextInput/styled";
import PropTypes from "prop-types";

const TextInput = ({
  label,
  errorMessage,
  onChange,
  value,
  name,
  ...props
}) => {
  return (
    <Container>
      {label && <Label>{label}</Label>}
      <Input
        name={name}
        onChange={onChange}
        type="text"
        value={value}
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...props}
      />
      {errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
    </Container>
  );
};

TextInput.propTypes = {
  label: PropTypes.string,
  errorMessage: PropTypes.string,
  name: PropTypes.string,
  onChange: PropTypes.func,
  value: PropTypes.string,
};

TextInput.defaultProps = {
  label: null,
  errorMessage: null,
  name: null,
  onChange: null,
  value: null,
};

export default TextInput;
