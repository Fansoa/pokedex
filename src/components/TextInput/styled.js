import styled from "@emotion/styled";
import global from "@/src/theme/global";
import pxToRem from "@/src/utils/methods/pxToRem";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  gap: ${pxToRem(8)};
`;

export const Label = styled.p`
  font-weight: bold;
  font-size: ${pxToRem(14)};
  color: ${global.colors.taupeGrey};
  text-transform: uppercase;
`;

export const Input = styled.input`
  width: ${pxToRem(280)};
  height: ${pxToRem(56)};
  border-radius: ${pxToRem(4)};
  border: ${pxToRem(1)} solid ${global.colors.gainsboro};
  color: ${global.colors.darkSlateGray};
  font-size: ${pxToRem(24)};
  font-weight: bold;
  padding-left: ${pxToRem(8)};
  &:focus-visible {
    outline: none;
  }
`;

export const ErrorMessage = styled.p`
  color: ${global.colors.red};
`;
