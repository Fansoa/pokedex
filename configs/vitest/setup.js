import { expect, afterEach } from "vitest";
import { cleanup } from "@testing-library/react";
import * as matchers from "@testing-library/jest-dom/matchers";
import { createSerializer } from "@emotion/jest";

expect.extend(matchers);
expect.addSnapshotSerializer(createSerializer());

afterEach(() => {
  cleanup();
});
