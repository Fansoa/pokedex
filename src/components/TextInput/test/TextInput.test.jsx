import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { expect } from "vitest";
import renderWithIntegrationProvider from "@/configs/vitest/customRender";
import TextInput from "@/src/components/TextInput";
import { act } from "react-dom/test-utils";

describe("TestComponent", () => {
  let container;

  it("TextInput value change when user enters text", async () => {
    container = renderWithIntegrationProvider(<TextInput />).container;
    const user = userEvent.setup();
    const input = screen.getByRole("textbox");
    await act(async () => {
      await user.type(input, "message here!");
    });

    expect(input).toBeInTheDocument();
    expect(input).toHaveValue("message here!");
    expect(container).toMatchSnapshot();
  });

  it("TextInput should display label", async () => {
    container = renderWithIntegrationProvider(
      <TextInput label="label" />,
    ).container;
    const label = screen.getByText(/label/i);

    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("TextInput should display error message", async () => {
    container = renderWithIntegrationProvider(
      <TextInput errorMessage="error message" />,
    ).container;
    const errorMessage = screen.getByText(/error message/i);

    expect(errorMessage).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
