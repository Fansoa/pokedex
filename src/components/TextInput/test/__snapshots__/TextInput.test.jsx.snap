// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TestComponent > TextInput should display error message 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.emotion-1 {
  width: 17.5rem;
  height: 3.5rem;
  border-radius: 0.25rem;
  border: 0.0625rem solid #DEDEDE;
  color: #264653;
  font-size: 1.5rem;
  font-weight: bold;
  padding-left: 0.5rem;
}

.emotion-1:focus-visible {
  outline: none;
}

.emotion-2 {
  color: red;
}

<div>
  <div
    class="emotion-0"
  >
    <input
      class="emotion-1"
      type="text"
      value=""
    />
    <p
      class="emotion-2"
    >
      error message
    </p>
  </div>
</div>
`;

exports[`TestComponent > TextInput should display label 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.emotion-1 {
  font-weight: bold;
  font-size: 0.875rem;
  color: #8F8F8F;
  text-transform: uppercase;
}

.emotion-2 {
  width: 17.5rem;
  height: 3.5rem;
  border-radius: 0.25rem;
  border: 0.0625rem solid #DEDEDE;
  color: #264653;
  font-size: 1.5rem;
  font-weight: bold;
  padding-left: 0.5rem;
}

.emotion-2:focus-visible {
  outline: none;
}

<div>
  <div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      label
    </p>
    <input
      class="emotion-2"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`TestComponent > TextInput value change when user enters text 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.emotion-1 {
  width: 17.5rem;
  height: 3.5rem;
  border-radius: 0.25rem;
  border: 0.0625rem solid #DEDEDE;
  color: #264653;
  font-size: 1.5rem;
  font-weight: bold;
  padding-left: 0.5rem;
}

.emotion-1:focus-visible {
  outline: none;
}

<div>
  <div
    class="emotion-0"
  >
    <input
      class="emotion-1"
      type="text"
      value=""
    />
  </div>
</div>
`;
